internal/controller/nicconfigurationtemplate_controller.go

```diff
@@ -172,6 +174,22 @@ func (r *NicConfigurationTemplateReconciler) Reconcile(ctx context.Context, req
 		}
 	}
 
+	err = syncdaemon.SyncConfigDaemonObjs(ctx, r.Client, r.Scheme, syncdaemon.ConfigDaemonParameters{
+		OwnerName:          os.Getenv("POD_NAME"),
```

Comment:
pod restart would trigger deletion and re-creation of the daemonset.

i think we need to use a new CR (NICConfigurationOperatorConfig) or a configmap

Reply:
Introduced a new config map with daemon configuration, now the ds refers to it as the owner